src/resolve/test-resolved-etc-hosts.c

```diff
@@ -37,6 +37,9 @@ static void test_parse_etc_hosts(void) {
         assert_se(f);
         fputs("1.2.3.4 some.where\n", f);
         fputs("1.2.3.5 some.where\n", f);
+        fputs("1.2.3.6 dash dash-dash.where-dash\n", f);
```

Comment:
hmm, why this patch? are you sure hostnames, with leading/trailing dashes are not OK? note that in dnssd and SRV RRs domain names with leading underscore are explicitly OK... 

What's you reasoning for this, is there any RFC or otherwise recommendation to disallow these?

Reply:
Dashes and underscores are ok in general, but if the entry has A/AAAA records (i.e. it's considered to be a "hostname") then many implementations – including Bind9 as reference – follow the more restrictive syntax from [RFC 1035 section 2.3.1](https://tools.ietf.org/html/rfc1035#section-2.3.1) which disallows leading/trailing dashes or underscores. For example, here's what `named-checkzone` says:

```
./test.zone:32: -ahost.example.com: bad owner name (check-names)
./test.zone:33: ahost-.example.com: bad owner name (check-names)
./test.zone:34: _ahost.example.com: bad owner name (check-names)
./test.zone:35: ahost_.example.com: bad owner name (check-names)
./test.zone:36: a_host.example.com: bad owner name (check-names)
```

Whereas if the entry only has "non-host" records (like SRV or TXT or PTR), then everything is allowed.